src/lib.ts

```
@@ -0,0 +1,362 @@
+
+function isSegmentSeparator(checkChar: string | undefined) {
+  if (!checkChar) return false
+  return ['/', '.', path.win32.sep].includes(checkChar)
+}
+
+function getParentRouteIds(
+  routeIds: string[],
+): Record<string, string | undefined> {
+  return routeIds.reduce<Record<string, string | undefined>>(
+    (parentRouteIds, childRouteId) => ({
+      ...parentRouteIds,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)